File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
  const last = arr.length - 1
  let maxIndex = -1
  for (let i = 0; i < last; i++) {
    if (maxIndex > i) {
      arr[i] = arr[maxIndex]
     } else {
       const sub = arr.slice(i + 1)
       const max = Math.max(...sub)
       maxIndex = i + 1 + sub.indexOf(max)
       arr[i] = max
     }
  }
  arr[last] = -1
  return arr
};

File: 1299-replace-elements-with-greatest-element-on-right-side/1299-replace-elements-with-greatest-element-on-right-side.go
func replaceElements(arr []int) []int {
  n := len(arr)
  max := arr[n-1]
  arr[n-1] = -1
  for i := n - 2; i >= 0; i-- {
    v := arr[i]
    arr[i] = max
    if v > max {
      max = v
    }
  }
  return arr
}